pkg/kube/kubehttp.go

```diff
@@ -376,6 +390,91 @@ func (k *KubeHTTP) SetResource(ctx context.Context, resource Resource) error {
 	return nil
 }
 
+func (k *KubeHTTP) SetWegoConfig(ctx context.Context, config WegoConfig, namespace string) error {
+	configBytes, err := kyaml.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("failed marshalling wego config: %w", err)
+	}
+
+	name := types.NamespacedName{Name: WegoConfigMapName, Namespace: namespace}
+
+	cm := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name.Name,
+			Namespace: name.Namespace,
+		},
+	}
+
+	if err := k.Client.Get(ctx, name, cm); err != nil {
+		if apierrors.IsNotFound(err) {
+			cm.Data = map[string]string{
+				"config": string(configBytes),
+			}
+
+			if err = k.Client.Create(ctx, cm); err != nil {
+				return fmt.Errorf("failed creating weave-gitops configmap: %w", err)
+			}
+
+			return nil
+		}
+
+		return fmt.Errorf("failed getting weave-gitops configmap: %w", err)
+	}
+
+	cm.Data["config"] = string(configBytes)
+
+	return k.Client.Update(ctx, cm)
+}
+
+func (k *KubeHTTP) GetWegoConfig(ctx context.Context, namespace string) (*WegoConfig, error) {
+	name := types.NamespacedName{Name: WegoConfigMapName, Namespace: namespace}
+
+	cm := &corev1.ConfigMap{}
+
+	if namespace != "" {
+		if err := k.Client.Get(ctx, name, cm); err != nil {
+			if apierrors.IsNotFound(err) {
+				return &WegoConfig{}, ErrWegoConfigNotFound
+			}
+
+			return nil, fmt.Errorf("failed getting weave-gitops configmap: %w", err)
+		}
+	} else {
+		configMap, err := k.getWegoConfigMapFromAllNamespaces(ctx)
```

Comment:
This is a bit confusing to me.  Typically no namespace implies "default" namespace.  This code pulls all configmaps and returns the first that has a matching name.  I think this behavior might be confusing if there are many configmaps across multiple namespaces - unlikely but likely causes additional issues.  I don't think we should propagate them here.  I think the code should either 
* if no namespace, use the default weave-gitops namespace 
* or return an error and require the caller to set the namespace

Reply:
If you do keep this behavior, it should be documented for the exported function.

Reply:
I totally agree is confusing, that came out from me trying to find a good way to search if there is a Wego Config in the cluster, since having just one by cluster is mandatory, the validation has been added.